### ISSUE TITLE
Type Error: NextApiRequest not assignable to Request | NextRequest | undefined

### ISSUE THREAD

**Issue by ericsen-tsai** [andreizanik/cookies-next#71]
**Description:**
Encountering a type error when using `NextApiRequest` and `NextApiResponse` in a Next.js API route handler. The error message indicates that `NextApiRequest` is not assignable to the type `Request | NextRequest | undefined`.

**Error Details:**
```
Type 'NextApiRequest' is not assignable to type 'Request | NextRequest | undefined'.
  Type 'NextApiRequest' is missing the following properties from type 'NextRequest': geo, ip, nextUrl, page, and 19 more.ts(2322)
```

**Code Snippet:**
```typescript
import { setCookie } from 'cookies-next';
import { NextApiRequest, NextApiResponse } from 'next';

...

export default async function handler(
  req: NextApiRequest,
  res: NextApiResponse<ClientServiceAuthResponse | ClientServiceAuthError>
) {
  try {
    const serverToken = await getServerToken();
    const clientToken = {
      ...
    };

    setCookie(TOKEN_COOKIE, clientToken.accessToken, {
      req,
      res,
      expires: new Date(Date.now() + clientToken.expiresIn),
    });
    res.status(200).json(clientToken);
  } catch (e) {
    res.status(500).json({ message: 'Internal Server Error' });
  }
}
```

**Steps to Reproduce:**
1. Implement the given API route handler in a Next.js project.
2. Run `tsc --project tsconfig.json`.
3. Observe the type error in the terminal output.

**Expected Behavior:**
The TypeScript types should match, and the project should compile without type errors.

**Actual Behavior:**
TypeScript throws a type error, indicating that `NextApiRequest` is not assignable to the expected type.

**Additional Context:**
- Next.js version: 14.1.4
- TypeScript version: 5.4.3

**Comment by andreizanik** [comment]
Hello! 
Looks like this bug related with latest version (v4.2.0)
Can you check v4.1.1 in your context?

**Comment by ericsen-tsai** [comment]
v4.1.1 works fine for me.
Just create a PR #72  to move @types/cookie back to dependencies and it looks good to me.

**Comment by andreizanik** [comment]
Thanks! 
I published a new version v4.2.1
